mlflow/pipelines/cards/__init__.py

```diff
@@ -252,17 +252,22 @@ def render_table(table, columns=None, hide_index=True):
                         ("text-align", "left"),
                         ("padding", "5px"),
                     ],
-                }
+                },
             ]
         )
         if hide_index:
-            return (
+            rendered_table = (
                 styler.hide(axis="index").to_html()
                 if pandas_version >= Version("1.4.0")
                 else styler.hide_index().render()
             )
         else:
-            return styler.to_html() if pandas_version >= Version("1.4.0") else styler.render()
+            rendered_table = (
+                styler.to_html() if pandas_version >= Version("1.4.0") else styler.render()
+            )
+        return (
+            "<iframe srcdoc='{src}' width='100%' height='500' frameborder='0'></iframe>"
```

Comment:
Not a HTML expert, but wouldn't this fix the table to 500px no matter how large the table is? i.e. what happens if the table has only 1 row?

Reply:
+1. @sunishsheth2009 @harupy can advise

Reply:
Is there a reason we need to create an iframe here?
Can we simply add a max-height here?
```
 styler = table.set_table_attributes('style="border-collapse:collapse"').set_table_styles(
            [
                {
                    "selector": "table, th, td",
                    "props": [
                        ("border", "1px solid grey"),
                        ("text-align", "left"),
                        ("padding", "5px"),
                        ("max-height", "500px")
                    ],
                },
            ]
        )
```

Reply:
@jinzhang21 As shown in test plan. The height is fixed to 500 for tables and I think that's desired behavior. I'll create a task to set all card heights to 500.

Reply:
Ahhh sorry, I missed the selectors there, can we try something like this:
```
 styler = table.set_table_attributes('style="border-collapse:collapse"').set_table_styles(
            [
                {
                    "selector": "table, th, td",
                    "props": [
                        ("border", "1px solid grey"),
                        ("text-align", "left"),
                        ("padding", "5px"),
                        ("max-height", "500px")
                    ],
                },
               {
                    "selector": "table",
                    "props": [
                        ("max-height", "500px")
                    ],
                },
            ]
        )
```

Reply:
@sunishsheth2009 Yeah I've tried that too. It didn't work:(

Reply:
@sunishsheth2009 Actually my previous try was extract "table" to a separate block instead of the way you suggested to have "table" in both. 
I just tried your suggestion and it still doesn't work...
<img width="2067" alt="Screen Shot 2022-10-03 at 11 05 16 AM" src="https://user-images.githubusercontent.com/12734110/193647633-8bd7c3cd-b101-4567-b1c4-bca315eca2a2.png">
<img width="792" alt="Screen Shot 2022-10-03 at 11 05 21 AM" src="https://user-images.githubusercontent.com/12734110/193647635-33775dbf-7051-455c-82ff-275accd7b7de.png">

Reply:
Setting all cards to a fixed length seems OK to me. The next question is whether 500 is the best param to go with. Can we vary a few screen resolutions and see if 500x height looks reasonable in all / most circumstances?

Reply:
Setting fixed length will be another change. 
This change is to set the table height same as image and profile, which is 500px.

Reply:
If you set the whole card to 500px height, do you still need this PR?

Reply:
Adding `display: block` to `tbody` seems to work.

```python
import pandas as pd
from pathlib import Path


df = pd.DataFrame({"test": range(1000)})

html = (
    df.style.set_table_attributes('style="border-collapse:collapse"')
    .set_table_styles(
        [
            {
                "selector": "table, th, td",
                "props": [
                    ("border", "1px solid grey"),
                    ("text-align", "left"),
                    ("padding", "5px"),
                ],
            },
            {
                "selector": "tbody",
                "props": [
                    ("display", "block"),
                    ("max-height", "500px"),
                    ("overflow-y", "auto"),
                ],
            },
        ]
    )
    .to_html()
)

Path("test.html").write_text(html)
```



https://user-images.githubusercontent.com/17039389/193951041-562f94b7-fd21-49c9-906a-bce16d897992.mov

Reply:
Wrapping with `div` also seems to work.

```python
import pandas as pd
from pathlib import Path


df = pd.DataFrame({"test": range(1000)})

html = (
    df.style.set_table_attributes('style="border-collapse:collapse"')
    .set_table_styles(
        [
            {
                "selector": "table, th, td",
                "props": [
                    ("border", "1px solid grey"),
                    ("text-align", "left"),
                    ("padding", "5px"),
                ],
            },
            # {
            #     "selector": "tbody",
            #     "props": [
            #         ("display", "block"),
            #         ("max-height", "500px"),
            #         ("overflow-y", "auto"),
            #     ],
            # },
        ]
    )
    .to_html()
)

div = """
<div style="max-height: 500px; overflow: scroll;">
    {}
</div>
""".format(
    html
)

print(div)

Path("test.html").write_text(div)
```


https://user-images.githubusercontent.com/17039389/193954834-26c1b07b-9931-4305-a35d-2cbb9dcaeddb.mov

Reply:
I definitely prefer the [styling approach](https://github.com/mlflow/mlflow/pull/6954/files#r987432842) adding empty divs.
Can we do that @mingyu89 ?

Thanks @harupy for the amazing suggestion. :)

Reply:
@sunishsheth2009 sure. I'll use the "tbody" selector. Looks cleaner.

@harupy Thanks. I tried "div" too and changed to use iframe because that's what we use for pandas profile. I didn't figure out how to do it without another wrapper.
<img width="740" alt="Screen Shot 2022-09-30 at 9 00 06 PM" src="https://user-images.githubusercontent.com/12734110/193394075-de6fb2b9-7ebf-46c1-81bf-6a84560f22a3.png">

@jinzhang21 It depends on how we want to do it. Looks like doing it for each type without a common container is more preferable.